src/cdp/domains/input.zig

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2023-2024  Lightpanda (Selecy SAS)
+//
+// Francis Bouvier <francis@lightpanda.io>
+// Pierre Tachoire <pierre@lightpanda.io>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+
+pub fn processMessage(cmd: anytype) !void {
+    const action = std.meta.stringToEnum(enum {
+        dispatchMouseEvent,
+    }, cmd.input.action) orelse return error.UnknownMethod;
+
+    switch (action) {
+        .dispatchMouseEvent => return dispatchMouseEvent(cmd),
+    }
+}
+
+// https://chromedevtools.github.io/devtools-protocol/tot/Input/#method-dispatchMouseEvent
+fn dispatchMouseEvent(cmd: anytype) !void {
+    const params = (try cmd.params(struct {
+        type: []const u8,
+        x: u32,
+        y: u32,
+    })) orelse return error.InvalidParams;
+
+    try cmd.sendResult(null, .{});
+
+    if (std.ascii.eqlIgnoreCase(params.type, "mousePressed") == false) {
+        return;
+    }
+
+    const bc = cmd.browser_context orelse return;
+    const page = bc.session.currentPage() orelse return;
+    const click_result = (try page.click(cmd.arena, params.x, params.y)) orelse return;
```

Comment:
according with https://chromedevtools.github.io/devtools-protocol/tot/Input/#method-dispatchMouseEvent we should dispatch the event corresponding to the type given in parameter.
So I think we shoule:
* call `page.click` if the event type is ` mousePressed` or `mouseReleased`
* add a `TODO` to dispatch only the corresponding event in case of `mouseMoved`, `mouseWheel` (and we don't want to click in this case)

Reply:
I made it all more generic, page.click -> page.mouseEvent, taking in a struct when an event type enum)